src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -390,6 +418,7 @@ def __init__(self, snowflake_plan: LogicalPlan, *, analyzer: "Analyzer") -> None
         for query in self._snowflake_plan.queries:
             if query.params:
                 self._query_params.extend(query.params)
+        self.source_plan = self._snowflake_plan.source_plan
 
```

Comment:
would it be better to add this as a `@property` to `Selectable` class.

Reply:
Good point, updated